lib/osf-components/addon/components/file-browser/add-new/upload/component.ts

```diff
@@ -71,7 +78,24 @@ export default class Upload extends Component<Args> {
 
     @action
     addedFile(_: any, __: any, file: any) {
-        this.uploading.pushObject(file);
+        const cache = new TrackedWeakMap();
+        const initialValue = file.status;
+        Object.defineProperty(file, 'status', {
+            get() {
+                let existingValue = cache.get(file);
+                if (!existingValue) {
+                    existingValue = initialValue;
+                    cache.set(file, existingValue);
+                }
+                return existingValue;
+            },
+            set(value) {
+                cache.set(file, value);
+            },
+        });
```

Comment:
`TrackedWeakMap`, as the name suggests, is a `WeakMap` that is `tracked`. The use case here is to track a particular property (`status`) of the `file` object whose lifecycle we have no control over. Otherwise, the `status` property is not tracked by the template. See discussion here:
https://discord.com/channels/480462759797063690/981254016145186836